TestTaintBasedEvictions is flaky

<!-- Please only use this template for submitting reports about failing tests in Kubernetes CI jobs -->

**Which jobs are failing**:
https://testgrid.k8s.io/sig-release-master-blocking#integration-master

**Which test(s) are failing**:
TaintBasedEvictions

**Since when has it been failing**:
probably since it was added in this PR? https://github.com/kubernetes/kubernetes/pull/81856

**Testgrid link**:
https://testgrid.k8s.io/sig-release-master-blocking#integration-master

**Reason for failure**:
```
I0930 12:37:26.944]     --- FAIL: TestTaintBasedEvictions/Taint_based_evictions_for_NodeNotReady_with_no_pod_tolerations (35.04s)
I0930 12:37:26.944]         taint_test.go:770: Failed to taint node in test 1 <node-0>, err: timed out waiting for the condition
```

**Anything else we need to know**:
/sig scheduling
/kind flake
/priority important-soon
/milestone v1.17

I think I also run into this on 13th Oct:
```
I1013 12:04:38.321]     --- FAIL: TestTaintBasedEvictions/Taint_based_evictions_for_NodeNotReady_and_200_tolerationseconds (35.07s)
I1013 12:04:38.321]         taint_test.go:770: Failed to taint node in test 0 <node-1>, err: timed out waiting for the condition
```
Full logs:
https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/78623/pull-kubernetes-integration/1183345872050589697/build-log.txt

These flakes are consistently (always) timeouts waiting on node taints:
```
--- FAIL: TestTaintBasedEvictions/Taint_based_evictions_for_NodeNotReady_with_no_pod_tolerations (34.97s)
        taint_test.go:770: Failed to taint node in test 1 <node-2>, err: timed out waiting for the condition
```

https://github.com/kubernetes/kubernetes/blob/master/test/integration/scheduler/taint_test.go#L769-L771

--- a/test/integration/scheduler/taint_test.go
+++ b/test/integration/scheduler/taint_test.go
@@ -733,6 +733,34 @@ func TestTaintBasedEvictions(t *testing.T) {
 					t.Errorf("Failed to create node, err: %v", err)
 				}
 			}
+
+			// Regularly send heartbeat event to APIServer so that the cluster doesn't enter fullyDisruption mode.
+			// TODO(Huang-Wei): use "NodeDisruptionExclusion" feature to simply the below logic when it's beta.
+			var heartbeatChans []chan struct{}
+			for i := 0; i < nodeCount; i++ {
+				heartbeatChans = append(heartbeatChans, make(chan struct{}))
+			}
+			for i := 0; i < nodeCount; i++ {
+				// Spin up <nodeCount> goroutines to send heartbeat event to APIServer periodically.
+				go func(i int) {
+					for {
+						select {
+						case <-heartbeatChans[i]:
+							return
+						case <-time.Tick(2 * time.Second):
+							nodes[i].Status.Conditions = []v1.NodeCondition{
+								{
+									Type:              v1.NodeReady,
+									Status:            v1.ConditionTrue,
+									LastHeartbeatTime: metav1.Now(),
+								},
+							}
+							updateNodeStatus(cs, nodes[i])
+						}
+					}
+				}(i)
+			}
+
 			neededNode := nodes[1]
 			if test.pod != nil {
 				test.pod.Name = fmt.Sprintf("testpod-%d", i)
@@ -759,6 +787,13 @@ func TestTaintBasedEvictions(t *testing.T) {
 				}
 			}
 
+			for i := 0; i < nodeCount; i++ {
+				// Stop the neededNode's heartbeat goroutine.
+				if neededNode.Name == fmt.Sprintf("node-%d", i) {
+					heartbeatChans[i] <- struct{}{}
+					break
+				}
+			}
 			neededNode.Status.Conditions = test.nodeConditions
 			// Update node condition.
 			err = updateNodeStatus(cs, neededNode)
@@ -787,6 +822,10 @@ func TestTaintBasedEvictions(t *testing.T) {
 					t.Fatalf("Error: %v, Expected test pod to be %s but it's %v", err, test.waitForPodCondition, pod)
 				}
 				cleanupPods(cs, t, []*v1.Pod{test.pod})
+			}
+			// Close all heartbeat channels.
+			for i := 0; i < nodeCount; i++ {
+				close(heartbeatChans[i])
 			}
 			cleanupNodes(cs, t)
 			waitForSchedulerCacheCleanup(context.scheduler, t)

